### PR TITLE
fix(proxy): better error when upstream proxy fails

### PR DESCRIPTION
**Pull request by manniL**
### 🔗 Linked issue

#722

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR should ensure that `proxyRequest`s fail gracefully instead of leading to an unhandled node error crashing the server. 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/proxy.ts [hunk 1]
```diff
@@ -61,16 +61,22 @@ export async function proxyRequest(
...
+        method,
+        body,
+        duplex,
+        ...opts.fetchOptions,
+        headers: fetchHeaders,
+      },
+    });
+  } catch {
+    event.node.res.statusCode = 502;
+    event.node.res.statusMessage = "Bad Gateway";
```
Comment (pi0) [inline src/utils/proxy.ts:77]: I think only we can assume only proxy had been failed and perhaps we can double check if there is not already set statusCode and message on error (that currently is being ingored)

Additionally, this perhaps can be implemented directly in `sendProxy` util for consistent behavior

Comment (manniL) [reply src/utils/proxy.ts:77]: Moving back to draft until implemented! Good idea 👍🏻

**Comment by manniL** [comment]
Updated to `sendProxy`. 

> I think only we can assume only proxy had been failed and perhaps we can double check if there is not already set statusCode and message on error (that currently is being ingored)

You mean an existing one in `event.node.statusCode` / `statusMessage` or in the error directly? If the latter, which values should I check?

**Review by pi0 — APPROVED** [review]
Thanks ❤️ (updated impl to still throw error to allow hooking into it but with `createError`)

**Pull request merged**